include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,353 @@
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Constructors
+    //! @{
+    //==============================================================================================
+
+    //! Constructs an empty container.
+    soa_vector(Allocator const& a = Allocator{}) : storage(a) {}
+
+    //! Constructs the container with `n` default-inserted instances of `Type`.
+    explicit  soa_vector(std::size_t n, Allocator const& a = Allocator{})
+            : soa_vector(n, value_type{}, a)
+    {}
+
+    //! Constructs the container with `n` copies of elements with `value` value.
+    soa_vector(std::size_t n, value_type v, Allocator const& a = Allocator{})  : storage(a,n)
+    {
+      eve::algo::fill(*this, v);
+    }
+
+    //! Constructs the container with the contents of the initializer list `l`.
+    soa_vector( std::initializer_list<Type> l, Allocator const& a = Allocator{})
+              : storage(a, l.size())
+    {
+      auto ptr = l.begin();
+      for(std::size_t i = 0;i < size(); ++i)
+      {
+        eve::write(*ptr++, begin()+i);
+      }
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Capacity
+    //! @{
+    //==============================================================================================
+    //! Returns the number of elements in the container
+    EVE_FORCEINLINE std::size_t size()  const noexcept { return storage.size_; }
+
+    //! Returns the number of elements that the container has currently allocated space for
+    EVE_FORCEINLINE std::size_t capacity()  const noexcept { return storage.capacity_; }
+
+    //! Requests the removal of unused capacity.
+    // EVE_FORCEINLINE void shrink_to_fit() noexcept
+    // {
+    //   kumi::for_each([](auto& m) { m.shrink_to_fit(); }, storage);
+    // }
+
+    //! Checks if the container has no elements
+    EVE_FORCEINLINE bool empty() const noexcept { return size() == 0ULL; }
+
+    //! @brief Increase the capacity of the vector to a value that's greater or equal to `n`.
+    //! If `n` is greater than the current capacity(), new storage is allocated and all iterators,
+    //! including the past-the-end iterator, and all references to the elements are invalidated.
+    //! Otherwise, the method does nothing.
+    //! @warning reserve() does not change the size of the vector.
+    //! @param n New capacity of the `eve::soa_vector`
+    void reserve(std::size_t n)
+    {
+      if(n > capacity())
+      {
+        auto const sz = size();
+        soa_vector that(n);
```

Comment:
This fills in 0s I think. You should create a storage not a vector

Reply:
Indeed

Reply:
Still valid

Reply:
You didn't checked latest version of this file.